index.js

```diff
@@ -1,14 +1,32 @@
 'use strict';
 
 var superagent = require('superagent');
+var getProxyForUrl = require('proxy-from-env').getProxyForUrl;
+
+// extend with Request#proxy()
+require('superagent-proxy')(superagent);
 
 module.exports = function (remoteUrl, callback) {
-    superagent.get(remoteUrl).buffer().end(function (err, resp) {
-        if (err) {
-            return callback(err);
-        } else if (resp.ok) {
-            return callback(null, resp.text);
-        }
-        callback(new Error('GET ' + remoteUrl + ' ' + resp.status));
-    });
+    var proxy_url = getProxyForUrl(remoteUrl);
+
+    if(proxy_url) {
+        superagent.get(remoteUrl).proxy(proxy_url).buffer().end(function (err, resp) {
```

Comment:
Looks like the callback function is the same in both cases. This is duplicated code that could be factored out.

Also, please respect coding conventions and/or run eslint. Thanks.